src/protocol/http/src/test/java/org/apache/jmeter/protocol/http/util/TestGraphQLRequestParamUtils.java

```diff
@@ -97,9 +102,21 @@ void testInvalidGraphQLContentType(String contentType) {
         assertFalse(GraphQLRequestParamUtils.isGraphQLContentType(contentType));
     }
 
-    @Test
-    void testToPostBodyString() throws Exception {
-        assertEquals(EXPECTED_POST_BODY, GraphQLRequestParamUtils.toPostBodyString(params));
+    static Stream<org.junit.jupiter.params.provider.Arguments> postBodyFieldNameAndJsonNodes() throws Exception {
+        final JsonNode expectedPostBodyJson = objectMapper.readTree(EXPECTED_POST_BODY);
+        final JsonNode actualPostBodyJson = objectMapper.readTree(
+                GraphQLRequestParamUtils.toPostBodyString(new GraphQLRequestParams(OPERATION_NAME, QUERY, VARIABLES)));
+        return Stream.of(
+                arguments(GraphQLRequestParamUtils.OPERATION_NAME_FIELD, expectedPostBodyJson, actualPostBodyJson),
+                arguments(GraphQLRequestParamUtils.VARIABLES_FIELD, expectedPostBodyJson, actualPostBodyJson),
+                arguments(GraphQLRequestParamUtils.QUERY_FIELD, expectedPostBodyJson, actualPostBodyJson));
+    }
+
+    @ParameterizedTest
+    @MethodSource("postBodyFieldNameAndJsonNodes")
+    void testToPostBodyString(String fieldName, JsonNode expectedNode, JsonNode actualNode) throws Exception {
+        assertEquals(expectedNode, actualNode,
+                "The value of the '" + fieldName + "' field doesn't match in " + actualNode);
```

Comment:
Please refine the test so it is clear what it does.

Currently, it looks like it duplicates the same data 3 times, and `fieldName` seems to be unused.
What is the purpose of having parameters here?

Reply:
It is for testing whether the `#toPostBodyString()` has the equivalent JSON data by asserting each field, `operationName`, `variables` and `query`. And the `fieldName` is used in the assert message to explain which field is possibly different from the expectation.

Reply:
Thanks! I've improved it a bit : a00a07f6aeca4a247625b9d4212ce35adb49a5a7